api-http/proposals/http-plugin.md

```diff
@@ -0,0 +1,186 @@
+# Proposal for Nodeos Endpoint IP and Port Configuration
+This is a proposal to allow setting the IP and port for each endpoint provided by nodeos independently, i.e. address [IS #965](https://github.com/AntelopeIO/leap/issues/965).
+## API category classification
```

Comment:
How do users discover what categories exist? This is especially important due to the behavior being "opt out": I need to know when I enable some plugin the exhaustive list categories I need to go in and potentially disable (it's possible this even changes on new releases, meaning in 4.0 I configured the entire set of categories for a plugin, but then in 5.0 a new category is introduced that becomes default-enabled)

I like how some of the categories are named after the plugin and then ro/rw: `chain_ro` and `chain_rw`, or `net_ro` and `net_rw`. These are "predictable" in that their name is shared with the plugin and there is a pretty clear delineation between danger zones. Something like `producer_ro`, `producer_rw`, and `snapshot` is more problematic since all three of those are from the same plugin.